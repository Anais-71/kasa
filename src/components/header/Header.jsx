import { NavLink } from 'react-router-dom';
import logo from '../../assets/logo.svg'
import './header.css'

function Header() {
    const title = 'Kasa'
    return (
        <div className='header'>
            <img src={logo} alt='Kasa' className='logo' />
            <div className='nav'>
                <NavLink to="/"
                    data-testid="navlink-home"
                  className={({ isActive, isPending }) =>
                  isPending ? "pending" : isActive ? "active" : ""}>
                 Accueil
                </NavLink>
                <NavLink to="/about"
                data-testid="navlink-about"
                  className={({ isActive, isPending }) =>
                  isPending ? "pending" : isActive ? "active" : ""}>
                A propos
                </NavLink>
            </div>
        </div>
    )
}
 
export default Header