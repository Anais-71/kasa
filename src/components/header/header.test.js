import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

describe('Given I am on a page', () => {
  beforeEach(() => {
    render(
      <Router>
        <Header />
      </Router>
    );
  });

  test('When the header component is being displayed, Then the logo should be present', () => {
    const logo = screen.getByAltText('Kasa');
    expect(logo).toBeInTheDocument();
  });  

  test('When the header component is being displayed, Then the home link should be present', () => {
    const copyrights = screen.getByText(/Accueil/i);
    expect(copyrights).toBeInTheDocument();
  });

  test('When the header component is being displayed, Then the about link should be present', () => {
    const copyrights = screen.getByText(/A propos/i);
    expect(copyrights).toBeInTheDocument();
  });

  test('The header component matches the snapshot', () => {
    const { asFragment } = render(<Router>
      <Header />
    </Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});