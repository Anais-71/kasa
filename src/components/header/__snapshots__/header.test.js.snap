// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given I am on a page The header component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <img
      alt="Kasa"
      class="logo"
      src="logo.svg"
    />
    <div
      class="nav"
    >
      <a
        aria-current="page"
        class="active"
        data-testid="navlink-home"
        href="/"
      >
        Accueil
      </a>
      <a
        class=""
        data-testid="navlink-about"
        href="/about"
      >
        A propos
      </a>
    </div>
  </div>
</DocumentFragment>
`;
